### DISCUSSION TITLE
The pscale CLI is taking an ungodly amount of time to respond

### DISCUSSION THREAD

**Discussion by iambillmccann** [planetscale/discussion#438] [Q&A] [answered]
I am running on an Ubuntu disto on the Windows 11 WSL (2). I have installed pscale 0.136.0 and 0.134.0 in troubleshooting this issue.

The issue is simple to describe, the CLI takes as long as five minutes to respond. This is true for all commands, including pscale --help and pscale --version.

When I run pscale auth login, I will eventually get an authorization page in my browser, but the token has already expired when I click to authorize.

The closest discussion I have found to my issue "Planetscale CLI hangs entire system on login (M1 Mac)" ... but my system is not locked and I eventually get a response.

**Comment by fooforge** [comment]
@iambillmccann Oof, that sounds like a terrible experience.

Neither `pscale --help` nor `pscale --version` open network connections so I suspect this isn't just another instance of #181, but rather an issue with your stack or a new issue entirely.

Before I go down a rabbit hole trying to reproduce this, would you please try installing `pscale` natively on Windows via Scoop and see how that goes? How to install Scoop is documented on [its website](https://scoop.sh), the instructions for installing `pscale` is documented [here](https://planetscale.com/docs/concepts/planetscale-environment-setup#windows-instructions).

If it absolutely needs to run through WSL2, to be able to reproduce this I'm going to need more information from you. Mainly, your WSL2 build, the Ubuntu release you have chosen, how you've installed `pscale`, and perhaps the output from `dpkg -l` just in case.

**Reply by iambillmccann** [reply]
It runs great when installed on Windows via Scoop.

**Reply by iambillmccann** [reply]
I don't know that the WSL2 has a separate build number, but I am running Windows 11 version 22H2 (OS Build 22621.1631). It's Windows Pro.

My distro is Ubuntu 20.04.5 LTS

The dpkg output is ...

```
 ~/PlanetScale  sudo dpkg -i pscale_0.136.0_linux_amd64.deb
[sudo] password for iambillmccann:
Selecting previously unselected package pscale.
(Reading database ... 88639 files and directories currently installed.)
Preparing to unpack pscale_0.136.0_linux_amd64.deb ...
Unpacking pscale (0.136.0) ...
Setting up pscale (0.136.0) ...
 ~/PlanetScale 
```

**Reply by iambillmccann** [reply]
pscale --version ran 2 minutes 13 seconds

`pscale version 0.136.0 (build date: 2023-04-05T18:22:00Z commit: bd872ad)`

**Reply by iambillmccann** [reply]
Is the CLI open source? Can I get to the repository and run the code locally in a development environment?

**Reply by jgreet** [reply]
The source is in our CLI repo: https://github.com/planetscale/cli

**Reply by mscoutermarsh** [reply, marked as answer]
We're tracking this problem here: https://github.com/planetscale/cli/issues/637

It has something to do with accessing gnome-keyring for your access token. Running `pkill gnome-keyring` may help.
